plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTypes.java

```diff
@@ -13,26 +13,100 @@
  */
 package io.trino.plugin.iceberg;
 
+import com.google.common.math.LongMath;
+import io.airlift.slice.Slice;
 import io.airlift.slice.Slices;
+import io.trino.plugin.iceberg.util.Timestamps;
 import io.trino.spi.type.DecimalType;
 import io.trino.spi.type.Decimals;
+import io.trino.spi.type.Int128;
+import io.trino.spi.type.LongTimestampWithTimeZone;
 import io.trino.spi.type.UuidType;
 import org.apache.iceberg.types.Type;
 import org.apache.iceberg.types.Types;
 
 import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.MathContext;
 import java.nio.ByteBuffer;
 import java.nio.CharBuffer;
 import java.util.UUID;
 
 import static io.airlift.slice.Slices.utf8Slice;
 import static io.trino.plugin.iceberg.util.Timestamps.timestampTzFromMicros;
 import static io.trino.spi.type.Timestamps.PICOSECONDS_PER_MICROSECOND;
+import static java.lang.Math.toIntExact;
+import static java.math.RoundingMode.UNNECESSARY;
 
 public final class IcebergTypes
 {
     private IcebergTypes() {}
 
+    /**
+     * Convert value from Trino representation to Iceberg representation.
+     *
+     * @param icebergType target type to convert to
+     * @param value Trino value native for Trino type corresponding to {@code icebergType}
+     * @see TypeConverter#toTrinoType
+     */
+    public static Object convertTrinoValueToIceberg(Type icebergType, Object value)
```

Comment:
Is this the same as `ExpressionConverter#getIcebergLiteralValue`?

Reply:
thanks for catching this. i didn't find that function, will refactor